compiler-core/src/build/module_loader.rs

```diff
@@ -49,26 +80,34 @@ where
         let artefact = name.replace("/", "@");
         let source_mtime = self.io.modification_time(&path)?;
 
-        let read_source = |name| Ok(Input::New(self.read_source(path, name, source_mtime)?));
-
         let meta = match self.read_cache_metadata(&artefact)? {
             Some(meta) => meta,
-            None => return read_source(name),
+            None => return self.read_source(path, name, source_mtime).map(Input::New),
         };
 
-        // If there's a timestamp and it's newer or the same than the source
-        // file modification time then we read the cached data.
+        let mut module = None;
```

Comment:
This looks a bit awkward, but it ensures that we do not read source module multiple times in case hash check fails and we need the source for the codegen branch later.